docs/explainer.md

```diff
@@ -160,4 +160,38 @@ let writableStream = tcpSocket.writable;
 tcpSocket.close();
 ```
 
+To simplify security analysis, an API for listening for incoming connections is not yet being proposed.
+
 ## UDP
+
+To simplify security analysis, the initial proposal only supports cases where the web app initiates communication with a remote host.
+
+```
+const options = {
+    remoteAddress: 'example.com',
+    remotePort: 7
+};
+
+try {
+  const udpSocket = await navigator.openUDPSocket(options);
+  doStuffWith(udpSocket);
+} catch (err) {
+  // handle error
+}
+```
+
+`send` returns a promise, that resolves when a message is sent:
+
+```
+let blob = ...;
+await udpSocket.send(blob);
+```
+
+The UDP socket is `async iterable`, so we can asynchronously iterate to read incoming messages:
+
+```
+for await (let [source, blob] of udpSocket) {
```

Comment:
The source here allows for the future possibility of a single UDP socket communicating with multiple end points.